token/program-2022/src/extension.rs

```diff
@@ -1,6 +1,185 @@
 //! Extensions available to token mints and accounts
 
-use solana_program::{clock::Epoch, program_option::COption, pubkey::Pubkey};
+use {
+    crate::state::{pack_coption_key, unpack_coption_key, Account, Mint, Multisig},
+    arrayref::{array_mut_ref, array_ref},
+    solana_program::{
+        clock::Epoch,
+        program_error::ProgramError,
+        program_option::COption,
+        program_pack::{IsInitialized, Pack, Sealed},
+        pubkey::Pubkey,
+    },
+    std::convert::{TryFrom, TryInto},
+};
+
+const TYPE_LEN: usize = 1;
+const LENGTH_LEN: usize = 4;
```

Comment:
How about making TYPE_LEN + LENGTH_LEN = 4 to better align the data.  A `u8` for the type feels like we might one day run out of room, and `u32` for length seems more than what we'd ever need. `u16` for both?

Reply:
Fine by me